File: src/main/scala/com/bimbr/choreo/activities/ChoreoActivity.java
package com.bimbr.choreo.activities;

import android.app.Activity;

import com.bimbr.choreo.app.ChoreoApplication;
import com.bimbr.choreo.model.Choreography;

/**
 * Functionality common to all Choreo activities.
 *
 * @author mmakowski
 */
public abstract class ChoreoActivity extends Activity {
    protected ChoreoApplication application() {
        return (ChoreoApplication) getApplication();
    }

    protected Choreography choreography() {
        return application().getChoreography();
    }
}

File: src/main/scala/com/bimbr/choreo/activities/ChoreoActivity.scala
package com.bimbr.choreo.activities

import android.app.Activity
import com.bimbr.choreo.app.ChoreoApplication
import com.bimbr.choreo.model.Choreography
import com.bimbr.choreo.TypedActivity

abstract class ChoreoActivity extends Activity with TypedActivity {
  protected def application: ChoreoApplication = getApplication.asInstanceOf[ChoreoApplication]

  protected def choreography: Choreography = application.getChoreography
}
